src/test/java/com/meilisearch/integration/TasksTest.java

```diff
@@ -0,0 +1,92 @@
+package com.meilisearch.integration;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.meilisearch.integration.classes.AbstractIT;
+import com.meilisearch.integration.classes.TestData;
+import com.meilisearch.sdk.Index;
+import com.meilisearch.sdk.Task;
+import com.meilisearch.sdk.utils.Movie;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("integration")
+public class TasksTest extends AbstractIT {
+
+    private TestData<Movie> testData;
+
+    @BeforeEach
+    public void initialize() {
+        this.setUp();
+        if (testData == null) testData = this.getTestData(MOVIES_INDEX, Movie.class);
+    }
+
+    @AfterAll
+    static void cleanMeiliSearch() {
+        cleanup();
+    }
+
+    /** Test Get Task */
+    @Test
+    public void testClientGetTask() throws Exception {
+        String indexUid = "GetClientTask";
+        Task response = client.createIndex(indexUid);
+        client.waitForTask(response.getUid());
+
+        Task task = client.getTask(response.getUid());
+        assertTrue(task instanceof Task);
+        assertNotNull(task.getStatus());
+        assertNotEquals("", task.getStatus());
+        assertNotNull(task.getStartedAt());
+        assertNotNull(task.getEnqueuedAt());
+        assertNotNull(task.getFinishedAt());
+        assertTrue(task.getUid() >= 0);
+    }
+
+    /** Test Get Tasks */
+    @Test
+    public void testClientGetTasks() throws Exception {
+        Task[] tasks = client.getTasks();
+
+        for (Task task : tasks) {
+            assertTrue(task instanceof Task);
+            client.waitForTask(task.getUid());
+            assertNotNull(task.getStatus());
+            assertNotEquals("", task.getStatus());
+            assertTrue(task.getUid() >= 0);
+        }
+    }
+
+    /** Test waitForTask */
+    @Test
+    public void testWaitForTask() throws Exception {
+        String indexUid = "WaitForTask";
+        Task response = client.createIndex(indexUid);
+        client.waitForTask(response.getUid());
+
+        Task task = client.getTask(response.getUid());
+        assertTrue(task.getUid() >= 0);
+        assertNotNull(task.getEnqueuedAt());
+        assertNotNull(task.getStartedAt());
+        assertNotNull(task.getFinishedAt());
+        assertEquals("succeeded", task.getStatus());
+
+        client.deleteIndex(task.getIndexUid());
+    }
+
+    /** Test waitForTask timeoutInMs */
+    @Test
+    public void testWaitForTaskTimoutInMs() throws Exception {
+        String indexUid = "WaitForTaskTimoutInMs";
```

Comment:
I really liked what you did here on these tests you used this concept http://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure/

Maybe you could apply to all of your tests? 🐑

Reply:
You mean all the test file or all the tests in this file?

Reply:
In all the tests you touched in this PR (not only in this file 🥇). No need to add in the untouched files, because it will generate a lot of changes :) (feel free to do that in this PR, or raise a new issue if you want to) ⚡